astropy/table/table.py

```diff
@@ -206,6 +229,10 @@ def __setitem__(self, item, value):
                              .format(item))
         super().__setitem__(item, value)
 
+    def _setitem_validated(self, item, value):
```

Comment:
I think I would have done this myself by adding a keyword argument to `__setitem__` (see below), but possibly this is the less-surprising way. Anyway, mentioning it only in case you think "Ah, that is actually nicer" - this solution is definitely fine too.
```
def __setitem__(self, item, value, validated=False):
    if not validated:
        validation code
    return super().__setitem__(item, value)
```

Reply:
Excellent, it didn't occur to me to change the signature of `__setitem__`.  Done.